src/sql/parts/profiler/contrib/profilerActions.ts

```diff
@@ -98,11 +101,29 @@ export class ProfilerPause extends Action {
 	}
 
 	public run(input: ProfilerInput): TPromise<boolean> {
-		this.enabled = false;
-		return TPromise.wrap(this._profilerService.pauseSession(input.id).then(() => {
-			input.state.change({ isPaused: true, isStopped: false, isRunning: false });
-			return true;
-		}));
+		if(!this._paused) {
+			return TPromise.wrap(this._profilerService.pauseSession(input.id).then(() => {
+				this.paused = true;
+				input.state.change({ isPaused: true, isStopped: false, isRunning: false });
+				return true;
+			}));
+		}
+		else {
+			return TPromise.wrap(this._profilerService.pauseSession(input.id).then(() => {
```

Comment:
the code in the else snippet looks a lot like the code snippet at line 105.  You could probably remove the if statement and use `this._paused` to set the state values to reduce having similar code blocks.